colcon_core/task/python/build.py

```diff
@@ -142,7 +148,9 @@ async def build(self, *, additional_hooks=None):  # noqa: D102
                     '--no-deps',
                 ]
                 if setup_py_data.get('data_files'):
-                    cmd += ['install_data']
+                    cmd += ['symlink_data']
+                    if rc is not None:
```

Comment:
Is `rc is not None` semantically equivalent to "we didn't install this / can't undo this so we're gonna force it"? 

It took some spelunking to reason this case out so comments might be welcome here as breadcrumbs for future contributors.

Reply:
I literally have a commit with comments about this in my local branch, and I was just waiting to update this PR to include the new Python build task tests :)

Here's the summary from `_undo_install`:
```
Undo a previously run 'install' command.

:returns: None if install was not previously detected, otherwise
          an integer return code where zero indicates success.
```

The idea is that if we're switching between install and develop, we'll need to overwrite symlinks with files or vice-versa, which must be forced. Symlinks overriding symlinks and files overriding files doesn't seem to need that flag. None of this is my choosing - it's the way `install_data` behaves so I made `symlink_data` behave in kind.

Reply:
Comments added in 367200d.